examples/provide.ram/receiver.ram.cpp

```diff
@@ -0,0 +1,55 @@
+#include <eosiolib/eosio.hpp>
+#include <eosiolib/print.hpp>
+#include <eosiolib/bandwith.hpp>
+
+
+struct sharedval{
+    uint64_t id;
+    std::string string;
+
+    uint64_t primary_key() const {return id;}
+};
+
+using valtable = eosio::multi_index<"sharedval"_n, sharedval>;
+
+class [[eosio::contract]] receiver : public eosio::contract {
+public:
+    receiver( eosio::name receiver, eosio::name code, eosio::datastream<const char*> ds) : eosio::contract(receiver, code, ds), values(_self, _self.value) {}
+
+    [[eosio::action]] void writeself(eosio::name user) {
+        require_auth(user);
+        eosio::print("Write values to the user memory: ", user);
+
+        values.emplace(user, [&] (auto& value) {
```

Comment:
In this case we need to save data into smart-contract memory, so use `_self`, not user.